node/tests/SharedTests.ts

```diff
@@ -1878,6 +1878,122 @@ export function runBaseTests<Context>(config: {
         config.timeout,
     );
 
+    it.each([ProtocolVersion.RESP2, ProtocolVersion.RESP3])(
```

Comment:
Please take each test scenario to internal function with a nice naming , and run all of them from here. It will be easier to follow the tests and the meaning of each scenario. 
Right now its like a very big function which is hard to follow.

Reply:
Make sure that you cover all combinations.